common/static/common/js/discussion/utils.js

```diff
@@ -444,19 +463,52 @@
             }
         };
 
-        DiscussionUtil.abbreviateHTML = function(html, minLength) {
+        DiscussionUtil.convertMath = function(element, alreadyEscaped) {
+            var elementContent = element.html();
+            if (alreadyEscaped){
+                // markdownWithHighlight requires unescaped text, so we must unescape it.
+                // safe-lint: disable=javascript-escape
+                elementContent = _.unescape(edx.HtmlUtils.HTML(elementContent));
+            }
```

Comment:
@robrap This entire block of code turned out to be an attempt to compensate for the `converter.makeHtml` quirk of stripping html tags in post body, but escaping them in code blocks. So this caused both of them to be preserved and properly escaped, but in turn introduced more bugs in profile view, responses and comments.

So, I decided to stop fighting `converter.makeHtml` and just allow it to behave like it was; that means any html tags (including `script`) outside of code blocks will be stripped. If it is a UX regression - let's have it fixed as a separate PR/ticket/etc., as current master behaves exactly like that.

Answering the question (just to prove I'm sane and had a good idea why it should work like that :)):

1. It was safe - it came from `DiscussionThreadShowView.prototype.renderTemplate`
2. Html tags in code blocks was double-escaped (see explanation above)
3. That was a bug in this implementation.
4. disable - template linter reports all `_.escape` and `_.unescape`. Call to `HtmlUtils.HTML` - honestly - no idea; it shouldn't be there, but somehow it did what I wanted even with it (probably it just called toString on it).